src/OrleansRuntime/Placement/PlacementDirectorsManager.cs

```diff
@@ -39,63 +43,90 @@ private IPlacementDirector ResolveDirector(PlacementStrategy strategy)
             return result;
         }
 
-        public async Task<PlacementResult> SelectOrAddActivation(
+		private IActivationSelector ResolveSelector(PlacementStrategy strategy, bool addIfDoesNotExist = false)
+		{
+			IActivationSelector result;
+			var strategyType = strategy.GetType();
+			if (!this.selectors.TryGetValue(strategyType, out result) && addIfDoesNotExist)
+			{
+				var directorType = typeof(IActivationSelector<>).MakeGenericType(strategyType);
+				result = (IActivationSelector)this.serviceProvider.GetRequiredService(directorType);
+				this.selectors[strategyType] = result;
+			}
+
+			return result ?? defaultActivationSelector;
```

Comment:
>Not very familiar with the code.

Me neither. :-) `result` can be null if `TryGetValue` returns `false`. That's the case when we fall back to the default  selector.